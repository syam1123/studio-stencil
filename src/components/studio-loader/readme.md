# studio-loader



<!-- Auto Generated Below -->


## Dependencies

### Used by

 - [studio-form](../studio-form)

### Graph
```mermaid
graph TD;
  studio-form --> studio-loader
  style studio-loader fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
