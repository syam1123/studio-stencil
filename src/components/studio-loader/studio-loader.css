:host {
  display: block;
}

@keyframes ldio-iibc247m6p-1 {
  0% { transform: rotate(0deg) }
 50% { transform: rotate(-45deg) }
100% { transform: rotate(0deg) }
}
@keyframes ldio-iibc247m6p-2 {
  0% { transform: rotate(180deg) }
 50% { transform: rotate(225deg) }
100% { transform: rotate(180deg) }
}
.ldio-iibc247m6p > div:nth-child(2) {
transform: translate(-15px,0);
}
.ldio-iibc247m6p > div:nth-child(2) div {
position: absolute;
top: 40px;
left: 40px;
width: 120px;
height: 60px;
border-radius: 120px 120px 0 0;
background: #f8b26a;
animation: ldio-iibc247m6p-1 1s linear infinite;
transform-origin: 60px 60px
}
.ldio-iibc247m6p > div:nth-child(2) div:nth-child(2) {
animation: ldio-iibc247m6p-2 1s linear infinite
}
.ldio-iibc247m6p > div:nth-child(2) div:nth-child(3) {
transform: rotate(-90deg);
animation: none;
}@keyframes ldio-iibc247m6p-3 {
  0% { transform: translate(190px,0); opacity: 0 }
 20% { opacity: 1 }
100% { transform: translate(70px,0); opacity: 1 }
}
.ldio-iibc247m6p > div:nth-child(1) {
display: block;
}
.ldio-iibc247m6p > div:nth-child(1) div {
position: absolute;
top: 92px;
left: -8px;
width: 16px;
height: 16px;
border-radius: 50%;
background: #e15b64;
animation: ldio-iibc247m6p-3 1s linear infinite
}
.ldio-iibc247m6p > div:nth-child(1) div:nth-child(1) { animation-delay: -0.67s }
.ldio-iibc247m6p > div:nth-child(1) div:nth-child(2) { animation-delay: -0.33s }
.ldio-iibc247m6p > div:nth-child(1) div:nth-child(3) { animation-delay: 0s }
.loadingio-spinner-bean-eater-4tm8yii7439 {
width: 200px;
height: 200px;
display: inline-block;
overflow: hidden;
background: #ffffff;
}
.ldio-iibc247m6p {
width: 100%;
height: 100%;
position: relative;
transform: translateZ(0) scale(1);
backface-visibility: hidden;
transform-origin: 0 0; /* see note above */
}
.ldio-iibc247m6p div { box-sizing: content-box; }
/* generated by https://loading.io/ */