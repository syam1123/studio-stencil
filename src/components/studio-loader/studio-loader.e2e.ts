import { newE2EPage } from '@stencil/core/testing';

describe('studio-loader', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<studio-loader></studio-loader>');

    const element = await page.find('studio-loader');
    expect(element).toHaveClass('hydrated');
  });
});
