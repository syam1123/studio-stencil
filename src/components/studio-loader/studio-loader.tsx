import { Component, ComponentInterface, h } from '@stencil/core';

@Component({
  tag: 'studio-loader',
  styleUrl: 'studio-loader.css',
  shadow: true,
})
export class StudioLoader implements ComponentInterface {
  render() {
    return (
      <div class="loadingio-spinner-bean-eater-4tm8yii7439"><div class="ldio-iibc247m6p">
      <div><div></div><div></div><div></div></div><div><div></div><div></div><div></div></div>
      </div></div>
    );
  }

}
