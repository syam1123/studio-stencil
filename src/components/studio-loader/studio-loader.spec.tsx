import { newSpecPage } from '@stencil/core/testing';
import { StudioLoader } from './studio-loader';

describe('studio-loader', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [StudioLoader],
      html: `<studio-loader></studio-loader>`,
    });
    expect(page.root).toMatchSnapshot()
  });
});
