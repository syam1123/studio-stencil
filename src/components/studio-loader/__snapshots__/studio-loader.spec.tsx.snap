// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`studio-loader renders 1`] = `
<studio-loader>
  <mock:shadow-root>
    <div class="loadingio-spinner-bean-eater-4tm8yii7439">
      <div class="ldio-iibc247m6p">
        <div>
          <div></div>
          <div></div>
          <div></div>
        </div>
        <div>
          <div></div>
          <div></div>
          <div></div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</studio-loader>
`;
